src/api.jl

```diff
@@ -0,0 +1,258 @@
+const _API_DOCS_SHARED = """
+    parse_file(input, schema=nothing, consume_ctx::C; kwargs...) where {C<:AbstractConsumeContext} -> Nothing
+
+    setup_parser(input, schema=nothing; kwargs...) -> (Bool, ParsingContext, ChunkingContext, Lexer)
+    parse_file(
```

Comment:
This is explained below and in the README.

Reply:
yah, i just don't think it fits _here_. I think it should be a separate docstring added to the multi-arg method is documented. Otherwise it appears when i do `? setup_parser` and i found it confusing.